### PR TITLE
Switch to GitHub Flow and tools.build

### PR DESCRIPTION
**Pull request by TimoKramer**
##### Bugfix
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Architecture Decision Record added if design changes necessary
- [ ] Formatting checked

##### Feature
- [ ] Implements an existing feature request. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [x] Documentation added
- [x] Architecture Decision Record added 
- [ ] Formatting checked

### CODE REVIEW

**Comment by TimoKramer** [comment]
Should be merged into a newly created main-branch!

File: build.clj [hunk 1]
```diff
@@ -0,0 +1,71 @@
+(ns build
+  (:require [clojure.tools.build.api :as b]
+            [deps-deploy.deps-deploy :as dd]
+            [borkdude.gh-release-artifact :as gh]))
+
+(def lib 'timokramer/datahike)
```
Comment (kordano) [inline build.clj:6]: should this be `io.replikativ/datahike`?

File: deps.edn [hunk 2]
```diff
@@ -62,5 +59,14 @@
            :format {:extra-deps {cljfmt/cljfmt {:mvn/version "0.8.0"}}
                     :main-opts ["-m" "cljfmt.main" "check"]}
 
-           :ffix {:extra-deps {cljfmt/cljfmt {:mvn/version "0.8.0"}}
-                  :main-opts ["-m" "cljfmt.main" "fix"]}}}
+           :ffix {:extra-deps {cljfmt/cljfmt {:mvn/version "0.7.0"}}
+                  :main-opts ["-m" "cljfmt.main" "fix"]}
+
+           :build {:deps {io.github.clojure/tools.build {:git/tag "v0.6.6" :git/sha "4d41c26"}
```
Comment (kordano) [inline deps.edn:65]: what would you propose as update strategy for these dependencies?

File: doc/adl/adr-004-github-flow.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+## Consequences
+
+When the PR is merged into the newly created `main`-branch we are from then on only branching off
+of main for feature- and bugfix-branches. `main` needs to be the new default branch where everyone
+is landing on by default. Everytime a PR is merged there will be a new release on GitHub with the
+patch-level set to the overall commit-count. To make a new minor or major release the number
+needs to be adjusted manually in the build.clj-file.
+
+This change needs to be announced properly so that everyone working on Datahike knows about this
+new way of working. In the future all replikativ-libraries will be switched so everyone needs to
```
Comment (kordano) [inline doc/adl/adr-004-github-flow.md:35]: Could we reuse the `build.clj` for all the other libraries to build the projects?

File: doc/contributing.md [hunk 4]
```diff
@@ -32,40 +36,24 @@ clj -M:ffix
...
 ### CircleCI
-On CircleCI unit and integration tests are run. Afterwards, Datahike is being compiled and
-released. The official CircleCI jdk-8 image is used and the secret key to release to
 Clojars needs to be stored as an environment variable on CircleCI. There needs to be a
 variable `CLOJARS_USERNAME` set to your Clojars username and a variable `CLOJARS_PASSWORD` set
 to the token that permits to deploy on clojars.
 
+### Git tags
+Each release gets a release entry in Github and a git tag to point to the merge commit
+made when merging a branch into `main`. The jar is appended to the Github-release.
```
Comment (kordano) [inline doc/contributing.md:55]: How would the process with Gtihub releases work? Is a draft created automatically or is this done by the pipeline? And is it possible to fill in the text in the release text?

**Review by kordano — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 5]
```diff
@@ -1,10 +1,16 @@
 version: 2.1
 
 orbs:
+  github-cli: circleci/github-cli@1.0
   tools: replikativ/clj-tools@0
```
[the remark this answers is not in the document]
Comment (TimoKramer) [reply .circleci/config.yml:5]: it always uses the latest 0.*-version here. Update strategy is to not break anything. This change only has local changes in the circleci.yaml so far... when we want to keep this we can adapt the clj-tools and release a breaking minor/major and switch gradually on all the projects.

File: .circleci/config.yml [hunk 6]
```diff
@@ -1,10 +1,16 @@
...
+
 jobs:
   setup:
     executor: tools/clojurecli
+    parameters:
+      setup_cljs:
```
[the remark this answers is not in the document]
Comment (TimoKramer) [reply .circleci/config.yml:11]: yes it does. we were using it on some things @groundedSAGE was working on.

File: build.clj (hunk 1 above)
Comment (TimoKramer) [reply build.clj:6]: right :)

File: deps.edn (hunk 2 above)
Comment (TimoKramer) [reply deps.edn:65]: I would propose always to use the latest...should be stable or what do you mean?

File: doc/adl/adr-004-github-flow.md (hunk 3 above)
Comment (TimoKramer) [reply doc/adl/adr-004-github-flow.md:35]: I guess you need a build.clj for every project/repo...we could switch to a monorepo though:)

Comment (TimoKramer) [reply doc/adl/adr-004-github-flow.md:35]: and we could make a lib of the common functionality but I guess this is overengineered.

File: doc/contributing.md (hunk 4 above)
Comment (TimoKramer) [reply doc/contributing.md:55]: There is a github-draft-release made by the release-step of the pipeline. We can fill the text in the release as well with e.g. the CHANGELOG.md though this is huge.

File: doc/adl/adr-004-github-flow.md (hunk 3 above)
Comment (TimoKramer) [reply doc/adl/adr-004-github-flow.md:35]: maybe **it is** a good idea... :thinking:

File: doc/contributing.md (hunk 4 above)
Comment (TimoKramer) [reply doc/contributing.md:55]: We could make it a non-draft-release though...What do you prefer?

**Comment by TimoKramer** [comment]
I would adapt the PR-template then to have a box to tick for updating the CHANGELOG.md?!

File: doc/contributing.md (hunk 4 above)
Comment (TimoKramer) [reply doc/contributing.md:55]: We will for now manually add release-notes to the release. Thinking about automating this soonish.

**Comment by TimoKramer** [comment]
Should be good, can you give it another review @kordano ?

**Comment by kordano** [comment]
> I would adapt the PR-template then to have a box to tick for updating the CHANGELOG.md?!

yes, that would be good in the future

File: build.clj [hunk 7]
```diff
@@ -0,0 +1,86 @@
+(ns build
+  (:require
+    [borkdude.gh-release-artifact :as gh]
+    [clojure.tools.build.api :as b]
+    [deps-deploy.deps-deploy :as dd]))
+
+
+(def lib 'replikativ/datahike)
```
Comment (kordano) [inline build.clj:8]: that's `io.replikativ/datahike`, have a look [here](https://clojars.org/io.replikativ/datahike)

File: build.clj [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+(def version (format "0.4.%s" (b/git-count-revs nil)))
+(def current-commit (gh/current-commit))
+(def class-dir "target/classes")
+(def basis (b/create-basis {:project "deps.edn"}))
+(def jar-file (format "target/%s-%s.jar" (name lib) version))
+
+
+(defn clean
+  [_]
```
Comment (kordano) [inline build.clj:17]: What's the reason behind the empty parameters?

**Review by kordano — CHANGES_REQUESTED** [review]

File: build.clj (hunk 7 above)
Comment (TimoKramer) [reply build.clj:8]: oh man, you're totally right, :+1:

File: build.clj (hunk 8 above)
Comment (TimoKramer) [reply build.clj:17]: there is a params-list passed and nil when not used

**Review by kordano — APPROVED** [review]

**Pull request merged**